src/workspacer.Bar/Widgets/TitleWidget.cs

```diff
@@ -5,44 +5,33 @@ namespace workspacer.Bar.Widgets
 {
     public class TitleWidget : BarWidgetBase
     {
-        public Color MonitorHasFocusColor { get; set; } = Color.Yellow;
+        #region Properties
+        public Color WindowHasFocusColor { get; set; } = Color.Yellow;
         public bool IsShortTitle { get; set; } = false;
+        public int? MaxTitleLength { get; set; } = null;
+        public bool ShowAllWindowTitles { get; set; } = false;
+        public string TitlePreamble { get; set; } = null;
+        public string TitlePostamble { get; set; } = null;
         public string NoWindowMessage { get; set; } = "No Windows";
-
-
-        public override IBarWidgetPart[] GetParts()
-        {
-            var window = GetWindow();
-            var isFocusedMonitor = Context.MonitorContainer.FocusedMonitor == Context.Monitor;
-            var multipleMonitors = Context.MonitorContainer.NumMonitors > 1;
-            var color = isFocusedMonitor && multipleMonitors ? MonitorHasFocusColor : null;
-
-            if (window != null)
-            {
-                if (!IsShortTitle)
-                {
-                    return Parts(Part(window.Title, color, fontname: FontName));
-                }
-                else
-                {
-                    var shortTitle = GetShortTitle(window.Title);
-                    return Parts(Part(shortTitle, color, fontname: FontName));
-                }
-            }
-            else
-            {
-                return Parts(Part(NoWindowMessage, color, fontname: FontName));
-            }
-        }
+        public Func<IWindow, Action> TitlePartClicked = ClickAction;
+        public Func<IWindow, object> OrderWindowsBy = (window) => 0;
+        #endregion
 
         public override void Initialize()
         {
-            Context.Workspaces.WindowAdded += RefreshAddRemove;
-            Context.Workspaces.WindowRemoved += RefreshAddRemove;
+            Context.Workspaces.WindowAdded += RefreshAdd;
+            Context.Workspaces.WindowRemoved += RefreshRemove;
             Context.Workspaces.WindowUpdated += RefreshUpdated;
             Context.Workspaces.FocusedMonitorUpdated += RefreshFocusedMonitor;
         }
 
+        #region Get Windows
+        private IEnumerable<IWindow> GetWindows(bool filterOnTitleFilled = true)
```

Comment:
`IEnumerable<T>` requires you to add a using statement on top for `System.Collections.Generic` :smile:

Reply:
I believe this is the singular cause of the build failure and we won't be able to merge until this is resolved.